test/cpp/api/module.cpp

```diff
@@ -115,6 +115,17 @@ TEST_F(ModuleTest, ReplaceModule) {
   ASSERT_EQ(model->l1.get(), model->named_modules()["l1"]->as<Linear>());
 }
 
+TEST_F(ModuleTest, UnregisterModuleThrowsForUnknownModuleName) {
```

Comment:
nit: since this tests for both good and bad case, we might want to just name it `UnregisterModule`